lib/Fmt_ast.ml

```diff
@@ -3756,16 +3801,22 @@ and fmt_label_declaration c ctx ?(last = false) decl =
         ( hvbox 3
             ( hvbox 4
                 ( hvbox 2
-                    ( hovbox 2
-                        ( fmt_mutable_flag ~pro:noop ~epi:(fmt "@ ") c
-                            pld_mutable
-                        $ fmt_if (Option.is_some global_attr_opt) "global_ "
-                        $ fmt_str_loc c pld_name $ fmt_if field_loose " "
-                        $ fmt ":" )
-                    $ fmt "@ "
-                    $ fmt_core_type c (sub_typ ~ctx pld_type)
+                    ( hvbox 0
+                        ( hvbox 2
+                            ( hovbox 2
+                                ( fmt_mutable_flag ~pro:noop ~epi:(fmt "@ ")
+                                    c pld_mutable
+                                $ fmt_if
+                                    (Option.is_some global_attr_opt)
+                                    "global_ "
+                                $ fmt_str_loc c pld_name
+                                $ fmt_if field_loose " " $ fmt ":" )
+                            $ fmt "@ "
+                            $ fmt_core_type c (sub_typ ~ctx pld_type)
+                            $ fmt_if_k has_modalities cmt_after_type )
+                        $ fmt_modalities c pld_modalities )
                     $ fmt_semicolon )
-                $ cmt_after_type )
+                $ fmt_if_k (not has_modalities) cmt_after_type )
```

Comment:
Can you explain what's happening here with the comment formatting and why?

Reply:
I updated this code to be simpler and also handle comments better. In particular, this has to do with profiles with the option `--break-separators=after`, which makes semicolons on label declarations be at the end of the line rather than the beginning of the next line. There is a weird work around that makes it so that, in this case, comments after the type of the label declaration are moved to *after* the semicolon. To preserve the intention of this work around, if there are modalities, we instead take the comment that appears after the modalities (if present), and move *that* to after the semicolon. I added a test file to demonstrate this behavior (`modes_cmts-break_separators_after.ml.ref`; if you diff this against `modes_cmts.ref`, you can see that the comment is moved correctly).